cmd/bfgd/bfgd.go

```diff
@@ -148,61 +153,100 @@ func HandleSignals(ctx context.Context, cancel context.CancelFunc, callback func
 
 	select {
 	case <-ctx.Done():
-	case s := <-signalChan: // First signal, cancel context.
+	case s := <-signalChan:
 		if callback != nil {
-			callback(s) // Do whatever caller wants first.
-			cancel()
+			callback(s)
 		}
+		cancel()
 	}
-	<-signalChan // Second signal, hard exit.
+	<-signalChan
+	log.Errorf("Received second termination signal, forcing exit.")
 	os.Exit(2)
 }
 
-func _main() error {
-	// Parse configuration from environment
-	if err := config.Parse(cm); err != nil {
-		return err
+// initializeLogger configures logging for the application.
+func initializeLogger(logLevel string) error {
+	if err := loggo.ConfigureLoggers(logLevel); err != nil {
+		return fmt.Errorf("failed to configure logger with level %s: %w", logLevel, err)
 	}
+	log.Infof("Logger initialized with level: %s", logLevel)
+	return nil
+}
 
-	loggo.ConfigureLoggers(cfg.LogLevel)
-	log.Infof(welcome)
-
-	pc := config.PrintableConfig(cm)
-	for k := range pc {
-		log.Infof("%v", pc[k])
-	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	go HandleSignals(ctx, cancel, func(s os.Signal) {
-		log.Infof("bfg service received signal: %s", s)
-	})
 
+// runServer starts the BFG server with the provided configuration.
+func runServer(ctx context.Context) error {
 	server, err := bfg.NewServer(cfg)
 	if err != nil {
 		return fmt.Errorf("create BFG server: %w", err)
 	}
 	if err = server.Run(ctx); !errors.Is(err, context.Canceled) {
-		return fmt.Errorf("bfg server terminated: %w", err)
+		return fmt.Errorf("server terminated unexpectedly: %w", err)
 	}
-
 	return nil
 }
 
-func init() {
-	version.Component = "bfgd"
-	welcome = "Hemi Bitcoin Finality Governor " + version.BuildInfo()
```

Comment:
The `init` function needs to be kept in order to initialise the `welcome` variable, and set `version.Component = "bfgd"`.